### PR TITLE
Added inputs for hit options

### PR DESCRIPTION
**Pull request by ficocelliguy**
Added inputs for hit options and set up basic config saving via query strings

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -3,7 +3,8 @@
 <head>
   <meta charset="UTF-8">
   <title>Path of Damage</title>
-  <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.6.4/angular.min.js"></script>
+  <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.6.4/angular.min.js"></script>
```
Comment (PsychicMuffin) [inline index.html]: why does it say this line is changed? did you mess with the whitespace somehow?

File: index.html [hunk 2]
```diff
@@ -68,8 +69,11 @@ <h5 ng-if="table.total !== undefined">
       <th>Phys (Ele) Taken</th>
     </tr>
     <tr ng-repeat="hit in hits">
-      <td>{{hit}}</td>
-      <td>{{calcDamage(hit)}}</td>
+      <td class="damageInput">
+        <input class="form-control input-sm" type="number" ng-model="hit.damage" ng-model-options="{debounce: 500}"
+             ng-change="stringifyUrlData()" id="hit.id">
```
Comment (PsychicMuffin) [inline index.html]: indention a bit off here

File: style.css [hunk 3]
```diff
@@ -29,6 +29,21 @@ div.damageTable {
...
 
+.damageTable .damageInput input {
+    width: 8em;
+    background-color: transparent;
+    font-size: 14px;
+}
+
+.damageTable table tbody td.damageInput,
+.damageTable table tbody td.damageOutput {
+  padding: 0;
```
Comment (PsychicMuffin) [inline style.css]: indentation

File: style.css [hunk 4]
```diff
@@ -29,6 +29,21 @@ div.damageTable {
...
+}
+
+.damageTable table tbody td.damageOutput {
+  padding-left: 5px;
```
Comment (PsychicMuffin) [inline style.css]: indentation

File: app.js [hunk 5]
```diff
@@ -135,4 +158,27 @@ app.controller('Damage', function ($scope) {
...
+    window.history.replaceState({},"", "?data=" + stringified)
+  }
+
+  function getQueryVariable(variable)
+  {
+     var query = window.location.search.substring(1);
+     var vars = query.split("&");
+     for (var i=0;i<vars.length;i++) {
+       var pair = vars[i].split("=");
+       if(pair[0] == variable){return pair[1];}
```
Comment (PsychicMuffin) [inline app.js]: should be triple equals

File: app.js [hunk 6]
```diff
@@ -60,7 +68,17 @@ app.controller('Damage', function ($scope) {
     }
   };
 
-  $scope.hits = [100, 500, 1000, 2000, 3000, 4000, 5000, 7500, 10000];
+  if ($scope.damageDefaults == null) {
```
Comment (PsychicMuffin) [inline app.js]: ===

File: app.js [hunk 7]
```diff
@@ -101,9 +119,14 @@ app.controller('Damage', function ($scope) {
         }
       }
     }
+    $scope.stringifyUrlData();
   };
 
   $scope.calcDamage = function (hit) {
+    if (hit == null || hit == 0) {
```
Comment (PsychicMuffin) [inline app.js]: ===

File: app.js [hunk 8]
```diff
@@ -1,5 +1,13 @@
 var app = angular.module('PathOfDamage', []);
 app.controller('Damage', function ($scope) {
+var data = getQueryVariable("data");
```
Comment (PsychicMuffin) [inline app.js]: indent

File: app.js [hunk 9]
```diff
@@ -60,7 +68,17 @@ app.controller('Damage', function ($scope) {
...
 
-  $scope.hits = [100, 500, 1000, 2000, 3000, 4000, 5000, 7500, 10000];
+  if ($scope.damageDefaults === null) {
+    $scope.damageDefaults = [100, 500, 1000, 2000, 3000, 4000, 5000, 7500, 10000, null, null];
+  }
+  $scope.hits = [];
+  for (var i = 0; i < $scope.damageDefaults.length; i++) {
+    $scope.hits.push({
+      damage: $scope.damageDefaults[i],
+      id: "hit" + i
```
Comment (PsychicMuffin) [inline app.js:80]: why do the hits need Ids? I see that you use them in the html as well, but not anywhere in the css?

**Review by PsychicMuffin — APPROVED** [review]

File: app.js (hunk 9 above)
Comment (ficocelliguy) [reply app.js:80]: If we decide to store only the modified ones or something I thought we'd use them.

**Pull request merged**